src/gmail.js

```diff
@@ -1695,12 +1695,10 @@ var Gmail_ = function(localJQuery) {
 
     api.helper.get.visible_emails_pre = function() {
         var page = api.get.current_page();
-        var url = window.location.origin + window.location.pathname + "?ui=2&ik=" + api.tracker.ik+"&rid=" + api.tracker.rid + "&view=tl&num=120&rt=1";
-        var start = $(".aqK:visible .Dj").find("span:first").text().replace(",", "").replace(".", "");
+        var url = window.location.origin + window.location.pathname + "?ui=2&ik=" + api.tracker.ik+"&rid=" + api.tracker.rid + "&view=tl&rt=1&num=" + api.get.emails_per_page();
+        var start = $(".aqK:visible .Dj").find("span.ts:first").text().replace(",", "").replace(".", "");
         if (start) {
```

Comment:
I know we're already pretty heavy on magic CSS-selectors, but you could just for the sake of review outline how this `.ts` selector changes the behaviour of the function in a way which is consistent with your other changes?

Is this a required change, or is this a bugfix for something else which just happened to get submitted in the same batch?